### PR TITLE
Optional settings

### PR DESCRIPTION
**Pull request by pburnsdata**
# Issues
#106 
#115 

- Tried to reduce the duplication of settings 
- Controls are getting a lil crowded - gonna try to pretty that up when I tackle #118

### CODE REVIEW

File: settings-schema.json [hunk 1]
```diff
@@ -69,31 +23,34 @@
...
+                    "value_col" : "sitename",
                     "label": "Site",
-                    "default_nesting": true
+                    "default_nesting": true,
+                    "role": "site_col"
                 },
                 {
-                    "settings_col": "id_col",
-                    "value_col" : null,
+                    "value_col" : "id_col",
```
Comment (samussiah) [inline settings-schema.json]: id_col instead of subjectnameoridentifier?

File: settings-schema.json [hunk 2]
```diff
@@ -69,31 +23,34 @@
...
                     "label": "Subject ID",
-                    "default_nesting": true
+                    "default_nesting": true,
+                    "role": "id_col"
                 },
                 {
-                    "settings_col": "visit_col",
-                    "value_col" : null,
+                    "value_col" : "visit_col",
```
Comment (samussiah) [inline settings-schema.json]: visit_col instead of folderinstancename?

File: settings-schema.json [hunk 3]
```diff
@@ -69,31 +23,34 @@
...
                     "label": "Folder",
-                    "default_nesting": false
+                    "default_nesting": false,
+                    "role": "visit_col"
                 },
                 {
-                    "settings_col": "form_col",
-                    "value_col" : null,
+                    "value_col" : "form_col",
```
Comment (samussiah) [inline settings-schema.json]: form_col instead of ecrfpagename?

File: settings-schema.json [hunk 4]
```diff
@@ -69,31 +23,34 @@
...
                     "label": "Form",
                     "default_nesting": false
```
Comment (samussiah) [inline settings-schema.json]: why doesn't this one have a role?

File: src/configuration/rendererSettings.js [hunk 5]
```diff
@@ -86,10 +78,36 @@ export default function rendererSettings() {
...
+            },
+            {
+                value_col: 'subset3',
+                label: 'Subset 3'
+            },
+            {
+                value_col: 'subjfreezeflg',
+                label: 'Subject Freeze Status',
+                multiple: true,
+                subject_export: true // I'd'like to add a check that these only change for subject or higher but not sure how i would do so
```
Comment (samussiah) [inline src/configuration/rendererSettings.js]: good idea.  maybe make a subjectnameoridentifier-subjfreezeflg set and verify all the lengths are 1.

File: src/configuration/syncSettings.js [hunk 6]
```diff
@@ -1,18 +1,14 @@
...
-    });
-
     // sort value_cols so that crfs come before query cols regardless of order in rendererSettings
     settings.value_cols.sort(function(a, b) {
         return a.type < b.type ? -1 : a.type > b.type ? 1 : 0;
     });
 
+    // Assign nest variables with specfic roles to specific settings
+    settings.nestings.map(function(d) {
+        if (typeof d.role != 'undefined') settings[d.role] = d.value_col;
```
Comment (samussiah) [inline src/configuration/syncSettings.js]: makes sense to me.  still wondering why form doesn't have a role.

File: src/configuration/syncSettings.js [hunk 7]
```diff
@@ -22,12 +18,26 @@ export default function syncSettings(settings) {
     //Define table column variables.
     settings.cols = d3.merge([['id'], settings.value_cols.map(d => d.col)]);
 
+    // Define nesting filters
```
Comment (samussiah) [inline src/configuration/syncSettings.js]: noice

File: settings-schema.json (hunk 4 above)
Comment (pburnsdata) [reply settings-schema.json]: Interestingly, the form column isn't used anywhere besides summarization so it doesn't need one - as far as the chart knows it's just another variable

**Comment by pburnsdata** [comment]
That's a really good point @samussiah. there may not be one... it seemed like too many options but maybe this wasn't the way to go - I'll think about this again

File: settings-schema.json (hunk 3 above)
Comment (pburnsdata) [reply settings-schema.json]: updated these all in next pr

**Comment by samussiah** [comment]
review changes addressed in #126

**Pull request merged**